src/ClaimService.Business/Features/Claims/Commands/Create/CreateClaimHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using LT.DigitalOffice.ClaimService.Business.Shared.Enums;
+using LT.DigitalOffice.ClaimService.DataLayer;
+using LT.DigitalOffice.ClaimService.DataLayer.Models;
+using LT.DigitalOffice.Kernel.Extensions;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ClaimService.Business.Features.Claims.Commands.Create;
+
+public class CreateClaimHandler : IRequestHandler<CreateClaimCommand, Guid>
+{
+  private readonly IDataProvider _provider;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+
+  public CreateClaimHandler(
+    IDataProvider provider,
+    IHttpContextAccessor httpContextAccessor)
+  {
+    _provider = provider;
+    _httpContextAccessor = httpContextAccessor;
+  }
+
+  public Task<Guid> Handle(CreateClaimCommand request, CancellationToken ct)
+  {
+    Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
```

Comment:
do you need this variable?

Reply:
yes

Reply:
we have discussed once that from function design perspective it's not recommended